src/NUnitFramework/framework/TestContext.cs

```diff
@@ -547,6 +548,83 @@ public int InconclusiveCount
 #endregion
         }
 
-#endregion
+        #endregion
+
+        #region Nested PropertyBagAdapter Class
+
+        /// <summary>
+        /// <see cref="PropertyBagAdapter"/> adapts an <see cref="IPropertyBag"/>
+        /// for consumption by the user.
+        /// </summary>
+        public class PropertyBagAdapter
+        {
+            private IPropertyBag _source;
+            
+            /// <summary>
+            /// Construct a <see cref="PropertyBagAdapter"/> from a source
+            /// <see cref="IPropertyBag"/>.
+            /// </summary>
+            public PropertyBagAdapter(IPropertyBag source)
+            {
+                _source = source;
+            }
+
+            /// <summary>
+            /// Get the first property at <paramref name="key"/>, if it can be found, otherwise
+            /// returns null.
+            /// </summary>
+            public string Get(string key)
+            {
+                return _source.Get(key)?.ToString() ?? null;
+            }
+
+            /// <summary>
+            /// Returns if <paramref name="key"/> is found in this
+            /// <see cref="PropertyBagAdapter"/>.
+            /// </summary>
+            public bool ContainsKey(string key)
+            {
+                return _source.ContainsKey(key);
+            }
+
+            /// <summary>
+            /// Returns an <see cref="IEnumerable{String}"/> of properties
+            /// at the <paramref name="key"/>.
+            /// </summary>
+            public IEnumerable<string> this[string key]
+            {
+                get
+                {
+                    var collection = new List<string>();
+                    foreach(var obj in _source[key])
+                    {
+                        collection.Add(obj.ToString());
+                    }
+
+                    return collection;
+                }
+            }
+
+            /// <summary>
+            /// Returns the count of elements at the <paramref name="key"/>.
+            /// </summary>
+            public int Count(string key)
+            {
+                return _source[key].Count;
+            }
+
+            /// <summary>
+            /// Returns an <see cref="ICollection{String}"/> of the property keys.
```

Comment:
Sadly XML crefs do not make it possible to specify closed generic types. All this is doing is aliasing `T` to `String`, as though they had picked  `String` rather than `T` as the name of the generic type parameter. (I forget, IntelliSense may display this as `T` anyway.)
I'd say `a collection` since the return type is already highlighted on the line above in the IntelliSense tooltip.